File: server/models/Session.js
const mongoose = require('mongoose');

const sessionSchema = new mongoose.Schema({
    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    title: { type: String, required: true },
    type: { type: String, enum: ['Lecture', 'Lab'], required: true },
    dayOfWeek: { type: String, required: true }, // e.g., 'Monday'
    startTime: { type: String, required: true }, // e.g., '10:00'
    duration: { type: Number, required: true }, // in hours (1 for Lecture, 2 for Lab)
    location: { type: String },
    attendance: [{
        date: { type: Date },
        status: { type: String, enum: ['Present', 'Absent', 'Cancelled'], default: 'Present' }
    }]
}, { timestamps: true });

module.exports = mongoose.model('Session', sessionSchema);

File: client/lib/models/session.dart
class Session {
  final String id;
  final String title;
  final String type; // 'Lecture' or 'Lab'
  final String dayOfWeek;
  final String startTime;
  final int duration;
  final String location;
  final List<Attendance> attendance;

  Session({
    required this.id,
    required this.title,
    required this.type,
    required this.dayOfWeek,
    required this.startTime,
    required this.duration,
    required this.location,
    required this.attendance,
  });

  factory Session.fromJson(Map<String, dynamic> json) {
    return Session(
      id: json['_id'],
      title: json['title'],
      type: json['type'],
      dayOfWeek: json['dayOfWeek'],
      startTime: json['startTime'],
      duration: json['duration'],
      location: json['location'] ?? '',
      attendance: (json['attendance'] as List<dynamic>?)
              ?.map((e) => Attendance.fromJson(e))
              .toList() ??
          [],
    );
  }
}

class Attendance {
  final DateTime date;
  final String status;

  Attendance({required this.date, required this.status});

  factory Attendance.fromJson(Map<String, dynamic> json) {
    return Attendance(
      date: DateTime.parse(json['date']),
      status: json['status'],
    );
  }
}
